### PR TITLE
Some corrections for integration.

### PR DESCRIPTION
**Pull request by julien-f**

### CODE REVIEW

**Comment by julien-f** [comment]
@Wescoeur Is this branch still relevant?

**Comment by Wescoeur** [comment]
@julien-f Yes this branch is necessary for the incremental backups integration.

File: src/xapi.js [hunk 1]
```diff
@@ -1026,7 +1026,7 @@ export default class Xapi extends XapiBase {
 
     let host
     let snapshotRef
-    if (isVmRunning(vm)) {
+    if (isVmRunning(vm) && !onlyMetadata) {
```
Comment (julien-f) [inline src/xapi.js]: A comment explaining why would be welcome.

File: src/xapi.js [hunk 2]
```diff
@@ -1514,9 +1514,21 @@ export default class Xapi extends XapiBase {
     )
   }
 
+  async connectVBD (vbdId) {
```
Comment (julien-f) [inline src/xapi.js]: Camel case.

File: src/xapi.js [hunk 3]
```diff
@@ -1514,9 +1514,21 @@ export default class Xapi extends XapiBase {
...
+    await this.call('VBD.plug', vbdId)
+  }
+
+  async disconnectVBD (vbdId) {
```
Comment (julien-f) [inline src/xapi.js]: Camel case.

File: src/xo.js [hunk 4]
```diff
@@ -900,39 +929,42 @@ export default class Xo extends EventEmitter {
...
-        ),
-        'finish'
-      )
+      const sourceStream = await xapi.exportVdi(currentSnapshot.$id, {
+        baseId: isFull ? undefined : base.$id,
+        format: Xapi.VDI_FORMAT_VHD
+      })
+
+      const targetStream = fs.createWriteStream(backupFullPath, { flags: 'wx' })
+      sourceStream.on('error', () => targetStream.emit('error'))
```
Comment (julien-f) [inline src/xo.js]: Forward the error parameter as well.

File: src/xo.js [hunk 5]
```diff
@@ -941,11 +973,11 @@ export default class Xo extends EventEmitter {
   }
 
   async _importVdiBackupContent (xapi, file, vdiId) {
-    const stream = fs.createReadStream(file)
-    const stats = await fs.stat(file)
+    const [ stream, length ] = await this._openAndwaitReadableFile(
+      file, 'VDI to import not found in this remote')
```
Comment (julien-f) [inline src/xo.js]: Bad indentation: the line of the `(` and the line of the `)` should be at the same level.

File: src/xo.js [hunk 6]
```diff
@@ -1060,14 +1092,13 @@ export default class Xo extends EventEmitter {
...
+    return `${directory}/${backupFormat}`
   }
 
   async _importVmMetadata (xapi, file) {
-    const stream = fs.createReadStream(file)
-    const stats = await fs.stat(file)
-
-    return await xapi.importVm(stream, stats.size, { onlyMetadata: true })
+    const [ stream, length ] = await this._openAndwaitReadableFile(
+      file, 'VM metadata to import not found in this remote')
```
Comment (julien-f) [inline src/xo.js]: Idem: bad indentation.

File: src/xo.js [hunk 7]
```diff
@@ -766,14 +766,35 @@ export default class Xo extends EventEmitter {
...
+    const backups = filter(files, xvaFilter)
+
+    // List delta backups.
+    const deltaDirs = filter(files, file => startsWith(file, 'vm_delta_'))
+
+    for (const deltaDir of deltaDirs) {
+      const files = await fs.readdir(`${path}/${deltaDir}`)
+      const deltaBackups = filter(files, xvaFilter)
+
+      Array.prototype.push.apply(backups, mapToArray(deltaBackups, deltaBackup => `${deltaDir}/${deltaBackup}`))
```
Comment (julien-f) [inline src/xo.js]: You can use `backups.push(...mapToArray(`.

**Pull request merged**